### PR TITLE
RUN-2271: Don't clean up in ~BaseAudioContext

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-2271#comment-fa1c87aa

### CODE REVIEW

File: third_party/blink/renderer/modules/webaudio/base_audio_context.cc [hunk 1]
```diff
@@ -101,6 +101,13 @@ BaseAudioContext::BaseAudioContext(Document* document,
       periodic_wave_triangle_(nullptr) {}
 
 BaseAudioContext::~BaseAudioContext() {
+  if (recordreplay::AreEventsDisallowed() &&
+      recordreplay::IsRecordingOrReplaying("leak-references",
+                                           "~BaseAudioContext")) {
```
Comment (bhackett1024) [inline third_party/blink/renderer/modules/webaudio/base_audio_context.cc:106]: Just call AreEventsDisallowed and pass in a subfeature.  There isn't a need to use two separate feature tests here.

**Review by bhackett1024 — APPROVED** [review]
Skipping this cleanup seems dangerous, can we just leak the object instead?

File: third_party/blink/renderer/modules/webaudio/base_audio_context.cc (hunk 1 above)
Comment (Domiii) [reply third_party/blink/renderer/modules/webaudio/base_audio_context.cc:106]: I'm just going with what I have learned previously. I guess I have not considered using `AreEventsDisallowed(const char* why)` before. Changing it now.

**Comment by Domiii** [comment]
Cancelling for now as explained [here](https://linear.app/replay/issue/RUN-2271#comment-5555c9bf).
Will re-approach this with a better fix if it re-appears.

**Pull request closed without merging**